unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -178,6 +189,7 @@ public UTF8String clone() {
 
   @Override
   public int compareTo(final UTF8String other) {
+    if (other == null) return 1;
```

Comment:
I think it should be able to sort columns with `null` values. The behavior should be compared to a database, shouldn't it?